src/app/dataset-view/additional-components/metadata-component/components/set-transform/set-transform.component.ts

```diff
@@ -0,0 +1,175 @@
+import { TransformInput } from "./../../../../../api/kamu.graphql.interface";
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    OnInit,
+} from "@angular/core";
+import { NgbModal, NgbModalRef } from "@ng-bootstrap/ng-bootstrap";
+import { MatTreeNestedDataSource } from "@angular/material/tree";
+import { DatasetService } from "src/app/dataset-view/dataset.service";
+import { BaseComponent } from "src/app/common/base.component";
+import { MaybeNull } from "src/app/common/app.types";
+import { DatasetSchema } from "src/app/interface/dataset.interface";
+import {
+    GetDatasetSchemaQuery,
+    SqlQueryStep,
+} from "src/app/api/kamu.graphql.interface";
+import { EditSetTransformService } from "./edit-set-transform..service";
+import { DatasetInfo } from "src/app/interface/navigation.interface";
+import { ActivatedRoute, ParamMap } from "@angular/router";
+import { requireValue } from "src/app/common/app.helpers";
+import ProjectLinks from "src/app/project-links";
+import { DatasetHistoryUpdate } from "src/app/dataset-view/dataset.subscriptions.interface";
+import { DatasetNode, SetTransFormYamlType } from "./set-transform.types";
+import { FinalYamlModalComponent } from "../final-yaml-modal/final-yaml-modal.component";
+import { TemplatesYamlEventsService } from "src/app/services/templates-yaml-events.service";
+import { AppDatasetCreateService } from "src/app/dataset-create/dataset-create.service";
+import { EditPollingSourceService } from "../add-polling-source/edit-polling-source.service";
+@Component({
+    selector: "app-set-transform",
+    templateUrl: "./set-transform.component.html",
+    styleUrls: ["./set-transform.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SetTransformComponent extends BaseComponent implements OnInit {
+    public inputDatasets = new Set([] as string[]);
```

Comment:
Can't you just use `new Set<string>()` ?

Reply:
Yes, i can